logserver.c

```diff
@@ -529,42 +777,53 @@ void pv_logserver_toggle(struct pantavisor *pv, const char *rev)
 		return;
 
 	if (pv_config_get_log_capture()) {
-		logserver_start(&logserver_g, rev);
+		logserver_start(rev);
 	} else
 		logserver_stop();
 }
 
 int pv_logserver_init()
 {
-	struct epoll_event ep_event;
-	char path[108]; // size of sockaddr_un sun_path
+	dl_list_init(&logserver_g.fdlst);
+	dl_list_init(&logserver_g.tmplst);
 
-	pv_paths_pv_file(path, PATH_MAX, LOGCTRL_FNAME);
-	logserver_g.sock_fd = logserver_open_socket(path);
-	logserver_g.epoll_fd = epoll_create1(0);
+	logserver_g.out.sfile = pv_config_get_log_server_output_single_file();
+	logserver_g.out.ftree = pv_config_get_log_server_output_file_tree();
+
+	errno = 0;
 
-	if (logserver_g.epoll_fd < 0 || logserver_g.sock_fd < 0) {
-		pv_log(DEBUG, "logserver_g epoll_fd = %d",
-		       logserver_g.epoll_fd);
-		pv_log(DEBUG, "logserver_g sock_fd = %d", logserver_g.sock_fd);
+	logserver_g.logsock = logserver_open_socket(LOGCTRL_FNAME, true);
+
+	if (logserver_g.out.sfile)
+		logserver_g.fdsock = logserver_open_socket(LOGFD_FNAME, true);
+
+	logserver_g.epfd = epoll_create1(0);
+
+	if (logserver_g.epfd < 0 || logserver_g.logsock < 0 ||
+	    (logserver_g.fdsock < 0 && logserver_g.out.sfile)) {
+		pv_log(DEBUG, "logserver_g epoll_fd = %d", logserver_g.epfd);
```

Comment:
should these logs be changed to WARN or ERROR?

Reply:
I will rearrange this function because some fd are more critical than others, for example, the epoll fd. So I will split that `if` and change the log level as appropriate

Reply:
Check 93c1fa8 to see the new implementation